src/uri/path.rs

```diff
@@ -256,6 +286,80 @@ impl fmt::Display for PathAndQuery {
     }
 }
 
+// ===== PartialEq / PartialOrd =====
+
+impl PartialEq for PathAndQuery {
+    #[inline]
+    fn eq(&self, other: &PathAndQuery) -> bool {
+        self.as_str() == other.as_str()
```

Comment:
Does `ByteStr` not have a `PartialEq` implementation that can just be used? Would let us skip the `is_empty` checks in both `as_str` calls.

Reply:
Aw, probably does. Would that just be `self.data == other.data`?

Reply:
Hm, is it possible to end up with that case?

But yea, looks like `ByteStr` does implement `PartialEq`, so it could just be `==`. If we can use that, it could take advantage of pointer comparisons if two `Authority`s were backed by the same `Bytes` buffer.